core/src/Revolution/modX.php

```diff
@@ -1724,7 +1724,7 @@ public function runProcessor($action = '', $scriptProperties = array(), $options
         // We're limiting to subclasses of Processor to prevent instantiating arbitrary classes
         if (static::isProcessorClass($action)) {
             /** @var Processor $processor */
-            $processor = new $action($this, $scriptProperties);
+            $processor = call_user_func_array([$action, 'getInstance'], [&$this, $action, $scriptProperties]);
```

Comment:
Perhaps `$action::getInstance($this, $action, $scriptProperties)` would be an additional improvement?

Reply:
`$this` is passed with the `&` in the `call_user_func_array`, it didn't work when passing it directly to `getInstance`, so I used the same notation as at the bottom of the method.

Reply:
`&` shouldn't be needed when passing directly to a method (cause objects are always passed as reference).. but fair enough.

Reply:
yup, figured that out :)